packages/seal/src/Adapter/AdapterFactory.php

```diff
@@ -88,6 +88,16 @@ public function parseDsn(string $dsn): array
             if (\str_contains($dsn, ':///')) {
                 // make DSN like loupe:///full/path/project/var/indexes parseable
                 $dsn = \str_replace(':///', '://' . $adapterName . '/', $dsn);
+            } elseif (\DIRECTORY_SEPARATOR === '\\') {
+                // might be Windows and contain an absolute path like loupe://C:\path\project\var\indexes which will fail when parse_url is used
+                $dsnParts = \explode('://', $dsn);
+
+                return [
+                    'scheme' => $dsnParts[0] ?? '',
+                    'host' => '',
+                    'path' => $dsnParts[1] ?? '',
+                    'query' => [],
```

Comment:
Is there any options were we can just convert the DSN into a format supported by `parse_url` else we would handle here also query and fragment parameter correctly:

```
loupe:///full/path/project/var/indexes?key=value#fragment
```

Reply:
Maybe we could modify the path to be like this:
```php
loupe:/C:\path\project\var\indexes?key=value#fragment

Array
(
    [scheme] => loupe
    [path] => /C:\path\project\var\indexes
    [query] => key=value
    [fragment] => fragment
)
```

That way, it would at least not fail nor consider the `C:\` directory as a host.
That way we'd only need to remove the first slash.

See: https://3v4l.org/kI6B3

```php
<?php

function printOutput($output)
{
    echo (is_array($output) ? print_r($output, true) : 'failing') . "\r\n\n";
}

$unixPath    = 'loupe://full/path/project/var/indexes?key=value#fragment';
$windowsPath = 'loupe://C:\path\project\var\indexes?key=value#fragment';


// Normal unix path

printOutput(parse_url($unixPath));

// this one does not work
printOutput(parse_url($windowsPath));


//however this would work if we modify it later on

$windowsPathModified = 'loupe:/C:\path\project\var\indexes?key=value#fragment';
printOutput(parse_url($windowsPathModified));
```

```
Array
(
    [scheme] => loupe
    [host] => full
    [path] => /path/project/var/indexes
    [query] => key=value
    [fragment] => fragment
)


failing

Array
(
    [scheme] => loupe
    [path] => /C:\path\project\var\indexes
    [query] => key=value
    [fragment] => fragment
)
```

Reply:
See 085fc7efdbbff987a1b65b5340feaa9164b85d48